Exercises/NoModules/Chapter 08/Soln8_08.cpp

```diff
@@ -0,0 +1,26 @@
+// Exercise 8_8 More efficient recursive version of function for x to the power n, n positive or negative
+// Based on Ex8_17.cpp
+#include <iostream>
+#include <iomanip>
+
+long double power(double x, int n);
+
+int main()
+{
+  for (int i {-3}; i <= 3; ++i)     // Calculate powers of 8 from -3 to +3
+    std::cout << std::setw(10) << power(8.0, i);
+
+  std::cout << std::endl;
+}
+
+// Recursive function to calculate x to the power n
+long double power(double x, int n)
+{
+  if (n == 0)     return 1.0;
+  else if (n < 0) return 1.0 / power(x, -n);
+  else if (n % 2) return x * power(x, n-1);     // x is odd
+  
+  // If we make it this far, x > 0 and even
```

Comment:
above you checked if n is even, not x

Reply:
No. I checked that it is odd... If a number is even, n % 2 equals zero, and thus false, remember? ;-)

Reply:
yes, but the comment says "x > 0 and even", shouldn't that be "n > 0 and even"?